### PR TITLE
fix: Correct redirect paths for Create and Cancel buttons in in Pods-List and Broker-Details pages

### PR DESCRIPTION
**Pull request by Msarawan**
- Updated FormView and YamlEditorView to include returnUrl prop
- Modified AddBroker component to extract and pass returnUrl from URL parameters
- Ensured proper navigation back to the previous page (Pods List or Broker Details) upon clicking Create, Apply and Cancel buttons

### CODE REVIEW

**Review by lavocatt — CHANGES_REQUESTED** [review]
Good work! I think that improving the logic of these buttons is very much needed. Since this is quite a critical piece of logic, I'd like to see some tests testing the various scenarios. That is, clicking on create and cancel from various locations and to ensure we do go back to the proper place as a result. Can you add these please?

**Review by gaohoward — DISMISSED** [review]

File: src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx [hunk 1]
```diff
@@ -35,10 +62,41 @@ describe('BrokerDetailsBreadcrumb', () => {
...
+      </MemoryRouter>,
+    );
+    await waitForI18n(comp);
+
+    fireEvent.click(screen.getByText(`broker ${name}`));
+    expect(navigate).toHaveBeenCalledWith(
+      `/k8s/ns/${namespace}/brokers/${name}`,
+    );
+  });
+
```
Comment (lavocatt) [inline src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx:98]: It feels like these tests aren't related to the PR, can you postpone them to another one please?

File: src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx [hunk 2]
```diff
@@ -71,4 +129,77 @@ describe('BrokerDetailsBreadcrumb', () => {
...
+    fireEvent.click(screen.getByTestId('broker-toggle-kebab'));
+    fireEvent.click(screen.getByText('edit_broker'));
+
+    expect(navigate).toHaveBeenCalledWith(
+      `/k8s/ns/${namespace}/edit-broker/${name}?returnUrl=${encodeURIComponent(
+        window.location.pathname,
+      )}`,
+    );
+  });
```
Comment (lavocatt) [inline src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx:153]: If this test is testing that the `apply` button of the yaml editor navigates correctly to some other palce, why is it in the `BrokerDetailsBreadcrumb.test.tsx` file and not in the `YamlEditorView.test.tsx` file ?

File: src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx [hunk 3]
```diff
@@ -71,4 +129,77 @@ describe('BrokerDetailsBreadcrumb', () => {
...
+    fireEvent.click(screen.getByText('delete'));
+
+    await waitFor(() => {
+      expect(k8sDelete).toHaveBeenCalledWith({
+        model: AMQBrokerModel,
+        resource: { metadata: { name, namespace: namespace } },
+      });
+      expect(navigate).toHaveBeenCalledWith(`/k8s/ns/${namespace}/brokers`);
+    });
+  });
```
Comment (lavocatt) [inline src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx:204]: These tests aren't related to the `apply` and `cancel` buttons navigation PR.

File: src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx [hunk 4]
```diff
@@ -25,8 +51,19 @@ describe('BrokerPodsBreadcrumb', () => {
...
+      <MemoryRouter>
+        <BrokerPodsBreadcrumb name={name} namespace={namespace} />
+      </MemoryRouter>,
+    );
+    await waitForI18n(comp);
+
+    fireEvent.click(screen.getByText('brokers'));
+    expect(navigate).toHaveBeenCalledWith(`/k8s/ns/${namespace}/brokers`);
+  });
+
```
Comment (lavocatt) [inline src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx:65]: These tests aren't related to the apply and cancel buttons navigation PR.

File: src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx [hunk 5]
```diff
@@ -52,4 +89,65 @@ describe('BrokerPodsBreadcrumb', () => {
...
+    fireEvent.click(screen.getByTestId('broker-toggle-kebab'));
+    fireEvent.click(screen.getByText('edit_broker'));
+
+    expect(navigate).toHaveBeenCalledWith(
+      `/k8s/ns/${namespace}/edit-broker/${name}?returnUrl=${encodeURIComponent(
+        window.location.pathname,
+      )}`,
+    );
+  });
```
Comment (lavocatt) [inline src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx:109]: Same, this is testing the behavior of the apply button, why is this test in the breadcrumb test file?

File: src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx [hunk 6]
```diff
@@ -52,4 +89,65 @@ describe('BrokerPodsBreadcrumb', () => {
...
+    fireEvent.click(screen.getByText('delete'));
+
+    await waitFor(() => {
+      expect(k8sDelete).toHaveBeenCalledWith({
+        model: AMQBrokerModel,
+        resource: { metadata: { name, namespace: namespace } },
+      });
+      expect(navigate).toHaveBeenCalledWith(`/k8s/ns/${namespace}/brokers`);
+    });
+  });
```
Comment (lavocatt) [inline src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx:152]: These tests aren't related to the apply and cancel buttons navigation PR.

**Review by lavocatt — CHANGES_REQUESTED** [review]
Thanks for working on adding the tests, that's good progress. Can you however only keep the tests relevant to the matter of the PR and also make sure they are correctly placed in the source tree? Atm I find them a bit confusing and it's hard to understand why tests testing apply and cancel buttons for the froms are getting placed in breadcrumb test files.

**Comment by Msarawan** [comment]
> Thanks for working on adding the tests, that's good progress. Can you however only keep the tests relevant to the matter of the PR and also make sure they are correctly placed in the source tree? Atm I find them a bit confusing and it's hard to understand why tests testing apply and cancel buttons for the froms are getting placed in breadcrumb test files.

sure, i will update.

File: src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx (hunk 3 above)
Comment (Msarawan) [reply src/brokers/broker-details/components/BrokerDetailsBreadcrumb/BrokerDetailsBreadcrumb.test.tsx:204]: yeah, that's true. as i am updated breadcrumbs component in this PR so thought of updating the test cases for all edge cases in this PR itself. 
but now, sure i will include these tests in a new PR.

File: src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx (hunk 6 above)
Comment (Msarawan) [reply src/brokers/broker-pods/components/BrokerPodsBreadcrumb/BrokerPodsBreadcrumb.test.tsx:152]: sure , i will move to new PR.

**Review by lavocatt — APPROVED** [review]
LGTM!

**Comment by Msarawan** [comment]
> LGTM!

Thankyou...
